### ISSUE TITLE
Bug: ttlib::cstr::make_relative resolves symlinks, breaking ttBld

### ISSUE THREAD

**Issue by KeyWorksRW** [KeyWorksRW/ttLib#183]
### Description:
The `std::filesystem::relative` function resolves symlinks. As long as the symlink is pointing to the same drive, this works okay. However, if the symlink for a directory is pointing to a different drive letter, the result is an empty string. For example:

```
    // assume ttlib is in c:/github/ttLib
    cd c:/github/myproject

    ttllib::cstr csLib("c:/github/ttLib/lib");
    ttlib::cwd cwd;
    csLib.make_relative(cwd);
```

In the above example, `csLib` would now be `../ttLib/lib` which is what we want. But now suppose you make that library a symlink to a RAMDISK set to drive letter R:

```
    mklink /J c:/github/ttLib/lib  R:\lib
```

Run the same code calling make_relative and `csLib` will now be empty.

As far as I can tell, there isn't any way to call `fs::relative()` that won't convert symlinks. This is a particularly nasty problem for **ttBld** because it needs to create a script file with a relative path that can be checked into git and used by other devs. Just because one person happened to create a symlink to the `lib` directory doesn't mean the symlink will be identical on everyone's machine. We absolutely _have_ to use a relative path since we have no way of knowing where the repository got installed.

Bottom line is that we need to roll our own `make_relative` function that will _not_ resolve symlinks.
